app/controllers/GoogleAuthTrait.scala

```diff
@@ -0,0 +1,73 @@
+package controllers
+
+import java.util
+import com.google.api.client.googleapis.auth.oauth2.{GoogleAuthorizationCodeTokenRequest, GoogleAuthorizationCodeFlow,
+GoogleCredential, GoogleIdToken, GoogleIdTokenVerifier}
+
+import com.google.api.client.http.javanet.NetHttpTransport
+import com.google.api.client.json.jackson2.JacksonFactory
+import com.google.api.services.youtube.YouTubeScopes
+import play.api.mvc.RequestHeader
+import utils.credentials
+import play.api.Play
+import play.api.Logger
+
+/*
+Common code relating to google auth
+ */
+trait GoogleAuthTrait {
+  val JSON_FACTORY = JacksonFactory.getDefaultInstance
+  val HTTP_TRANSPORT = new NetHttpTransport()
+  val APP_CREDENTIALS_FILE = "/etc/credentials.json"
+  val SESSION_ID_KEY = "livestream-sessid"
+  val OAUTH2_CALLBACK_PATH = routes.GoogleAuthController.oauth2callback().toString()
+  val NEXT_PAGE_KEY = "next-page"
+  val ACCESS_TOKEN_KEY = "actok"
+
+  def oauth2_redirect_url: String = {
```

Comment:
OK let's look at how to get a long-lived token onto the server.  I generally prefer more fine-grained user access control - but it's a fair point that it may become unmanageable and also that users can break stuff if they have full youtube access (which it looks like you'd need to give them)